cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -442,6 +443,26 @@ impl MachInstEmit for Inst {
                     sink,
                 );
             }
+            &Inst::UExtend {
+                rd,
+                rs,
+                ty_in,
+                ty_out,
+            } => {
+                // TODO: support other types cast
+                if ty_in == I32 && ty_out == I64 {
+                    let rs = allocs.next(rs);
+                    let rd = allocs.next_writable(rd);
+
+                    debug_assert_eq!(rs, e0());
```

Comment:
Let's add assert that the output register is "A"

Reply:
This is a good note, thanks!